core/src/connection/pool.rs

```diff
@@ -580,7 +580,7 @@ where
                     let num_established =
                         if let Some(conns) = self.established.get_mut(connected.peer_id()) {
                             conns.remove(&id);
-                            conns.len()
+                            conns.len() as u32
```

Comment:
It seems to me that the only scenario where this may realistically result in a lossy cast is one where we have a bug in the connection management / counting, so it may be worthwhile triggering a panic?